docs/userguide/revealing.md

```diff
@@ -1,42 +1,122 @@
+largest fragment number present in the slide is `4`, so the slide will have four fragments in total.
 
+## Fragment placeholders
+If you have a lot of fragments on a slide, it might be tedious to count the fragment numbers
+manually and keep them in sync when you make changes to the slide. To make this process easier,
+you can use placeholders to refer to the current fragment.
+
+The *current* fragment is the fragment with the largest number defined on a slide so far. You can
+access its number using the [`current_step`](elsie.box.Box.current_step) method. This value changes
+dynamically as you add more fragments to the slide:
 ```python
-@elsie.slide()
-def steps(slide):
-    slide.box().text("Box 1")
-    slide.box(show="next+").text("Box 2")
-    slide.box(show="next+").text("Box 3")
+# slide.current_step() == 1
+slide.box().text("Box 1")
+# slide.current_step() == 1
+slide.box(show="2").text("Box 1")
+# slide.current_step() == 2
+slide.box(show="3").text("Box 1")
+# slide.current_step() == 3
+```
+
+To leverage the *current* fragment, you can use two placeholders in place of a fragment number
+passed to `show`:
+
+- The `"last"` placeholder resolves to the value of the *current* fragment.
+- The `"next"` placeholder increments the *current* fragment and then resolves to the incremented
+value.
+
+Using the placeholders, you can show boxes in the same order as they appear in the source code.
+Here is an example of their usage:
+```elsie
+slide.box().text("Box 1")
+
+# Moves current fragment to 2 and shows the box in fragment 2
+slide.box(show="next").text("Box 2")
+
+# Shows the box in fragment 2
+slide.box(show="last").text("Box 3")
+
+# Moves current fragment to 3 and shows the box in fragment 3
+slide.box(show="next").text("Box 4")
+```
+
+You can also combine the placeholders with open fragment ranges:
+```elsie
+slide.box().text("List item 1")
+slide.box(show="next+").text("List item 2")
+slide.box(show="last+", p_left=20).text("Subitem 1")
+slide.box(show="last+", p_left=20).text("Subitem 2")
+slide.box(show="next+").text("List item 3")
+slide.box(show="last+", p_left=20).text("Subitem 1")
+```
+
+## Overlaying boxes
+As you may have already noticed, using `show` only influences the visibility of a box, not its
+layout. Therefore, the space for a box is allocated in each fragment, even if the box is not visible
+in that fragment (if you know CSS, fragment visibility behaves like `visibility: hidden`,
+not `display: none`).
+
+If you instead want to show boxes gradually on top of one another, you can use the
+[`overlay`](elsie.boxmixin.BoxMixin.overlay) shortcut to render a box at the same location and size
+as another box.
+```elsie
+slide.overlay(show="1").text("Box 1")
+slide.overlay(show="2").text("Box 2")
+slide.overlay(show="3").text("Box 3")
+```
+
+When using overlays, make sure that the box that you are overlaying is actually visible in the
+fragment when you want it to be overlaid. For example, this will not work as expected:
```

Comment:
I would add "as you MAY expected", because you read carefully it should not surprise you:)